torch/_torch_docs.py

```diff
@@ -7859,7 +7859,7 @@ def merge_dicts(*dicts):
 
 add_docstr(torch.sort,
            r"""
-sort(input, dim=-1, descending=False, *, out=None) -> (Tensor, LongTensor)
+sort(input, dim=-1, descending=False, stable=False, *, out=None) -> (Tensor, LongTensor)
```

Comment:
So let's talk about the plan for signatures.

Currently stable is only usable when dim, descending, and stable are provided as kwargs, right? Which means sort actually has two signatures in this PR:

```
sort(input, dim=-1, descending=False, *, out=None)
sort(input, *, dim=-1, descending=False, stable=False, out=None)
```

Which is confusing and would be tricky to document.

I think what we want to do is NOT DOCUMENT the stable flag for now, but file an issue (and link the issue in native_functions.yaml) that in one month we want to merge the signatures in native_functions.yaml. That is, in native_functions.yaml with this PR we have:

```
func: sort.values(Tensor self, int dim=-1, bool descending=False, *, Tensor(a!) values, Tensor(b!) indices) -> (Tensor(a!) values, Tensor(b!) indices)

func: sort.values_stable(Tensor self, *, bool? stable, int dim=-1, bool descending=False, Tensor(a!) values, Tensor(b!) indices) -> (Tensor(a!) values, Tensor(b!) indices)
```

And in one month I think we can update it to:

```
func: sort.values(Tensor self, int dim=-1, bool descending=False, bool? stable, *, Tensor(a!) values, Tensor(b!) indices) -> (Tensor(a!) values, Tensor(b!) indices)
```

And then update the documentation.

The reason we're adding a new signature (for now) is for forward compatibility. That is, if someone has a version of PyTorch with this PR and they use sort, we want them to match the OLD signature so when they export their model it can be loaded on a version of PyTorch without this PR. In a month, however, we can assume that everyone has updated PyTorch and we can match either signature. If we combine the signatures (like the above does), then I think the new combined signature will match this PR's kwarg-only signature, and we'll avoid a forward-compatibility problem and have only one easy to document signature.

Does that make sense?

Reply:
@mruberry 
This make sense. 
I have some thought on the changed document. Given 1.8 is already cut, can we keep the changed document (with stable flag one)? We will merge the two versions soon (before 1.9), this doc might cause confusion in next 1 month (we can make this shorter), but not very long. The worst case is that when people use the stable flag but not given the other two keywords, they will get an error and the correct usage will be shown in the error message. I we keep it,  @nikitaved don't have to revert this doc and re-write the same thing again later. What do you think?

Reply:
If that's easiest I guess that's OK? Remaking the same edits in the follow-up PR also seems easy to me.

Reply:
I do not mind either approach.  But if I do not have to do anything extra in this PR, it's even better.

Reply:
Let's leave it. And merge the two sort apis within a month.